src/Voron/Platform/Pal.cs

```diff
@@ -75,7 +73,74 @@ static Pal()
             Int32 errorCode,
             void* sb,
             Int32 capacity,
-            out Int32 specialErrnoCodes
-            );
+            out Int32 specialErrnoCodes);
+
+        [DllImport(LIBRVNPAL, SetLastError = true)]
+        public static extern Int32 rvn_create_and_mmap64_file(
+            string filename,
+            Int64 initialFileSize,
+            PalFlags.MMAP_OPTIONS flags,
+            out void* handle,
+            out void* baseAddress,
+            out Int64 actualFileSize,
+            out Int32 errorCode);
+
+        [DllImport(LIBRVNPAL, SetLastError = true)]
+        public static extern Int32 rvn_prefetch_virtual_memory(
+            void *virtualAddress,
+            Int64 length,
+            out Int32 errorCode);
+
+        [DllImport(LIBRVNPAL, SetLastError = true)]
+        private static extern Int32 rvn_get_system_information(
+            out SystemInformation systemInformation,
+            out Int32 errorCode);
+
+        [DllImport(LIBRVNPAL, SetLastError = true)]
+        public static extern Int32 rvn_memory_sync(
+            void *address,
+            Int64 size,
+            PalFlags.MSYNC_OPTIONS flags,
+            out Int32 errorCode);
+
+        [DllImport(LIBRVNPAL, SetLastError = true)]
+        public static extern Int32 rvn_dispose_handle(
+            string filepath,
+            void* handle,
+            Int32 deleteOnClose,
+            out Int32 errorCodeClose,
+            out Int32 errorCodeUnlink);
+
+        [DllImport(LIBRVNPAL, SetLastError = true)]
+        public static extern Int32 rvn_unmap(
+            void* address,
+            Int64 size,
+            Int32 deleteOnClose,
+            out Int32 errorCodeUnmap,
```

Comment:
here it makes sense to leave one error code if madvise error should be ignored and maybe even not returning madvise error at all if we are not logging it. Is this still the case? why not leaving the option to read all the failures? (if it is a signature issue, we can implement `(Int32 *)errorCodeArray`)

Reply:
Because you are baking the implementation behavior directly into the interface.
What if we have a new behavior, we want to not have to change the signature.
For example, if we decided that encrypted files need to be zeroed first, and that fails, do we now need _another_  error code?

That violates design guidelines and something we want to avoid.